File: leetcode/3094-minimum-number-of-operations-to-make-array-empty/minimum-number-of-operations-to-make-array-empty.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        size_t counter = 0;
        unordered_map<int, size_t>mapping{};
        for_each(nums.begin(), nums.end(), [&mapping](const int& x) {
            mapping[x]++;
        });
        
        for (const auto& [x, y] : mapping) {
            if (y == 1) return -1;
            else counter += (y + 2) / 3;
        }

        return counter;
    }
};

File: leetcode/3094-minimum-number-of-operations-to-make-array-empty/minimum-number-of-operations-to-make-array-empty.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        num_counts = Counter(nums)
        cnt = 0
        for num, count in num_counts.items():
          if count == 1:
            return -1
          else:
            cnt = cnt + ((count + 2) // 3)
        

        return cnt 

        